### PR TITLE
resolve #1578 | update curator to latest version

### LINKED ISSUE #1578: Bump apache curator version in hermes
currently hermes is using version [2.12.0](https://github.com/allegro/hermes/blob/master/build.gradle#L64), which was released Mar 01, 2017, newer versions have been released since then, so we would like hermes to use the newer one - l[ink to mvn repository](https://mvnrepository.com/artifact/org.apache.curator/curator-framework)

### PR DESCRIPTION
**Pull request by kirsteend**
This change also removes the `Reaper` class that is no longer required since v2.9
see: https://www.javadoc.io/doc/org.apache.curator/curator-recipes/4.2.0/org/apache/curator/framework/recipes/locks/Reaper.html

Fixes #1578

### CODE REVIEW

**Review by szczygiel-m — APPROVED** [review]

**Review by szczygiel-m — CHANGES_REQUESTED** [review]
Hey @kirsteend can You please check the integration test logs? I see that there are a lot of warnings that there are no nodes in zookeeper: 
image

**Comment by kirsteend** [comment]
@szczygiel-m I'm having issues running the tests locally and don't have time to look into this for the next couple weeks. If this is something you folks want to get in sooner please feel free to reassign the issue to someone else.

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/admin/zookeeper/ZookeeperAdminTool.java [hunk 1]
```diff
@@ -47,10 +35,8 @@ public void retransmit(SubscriptionName subscriptionName) {
     private void executeAdminOperation(SubscriptionName subscriptionName, String name) throws Exception {
         String path = zookeeperPaths.adminOperationPath(name);
 
-        String createdPath = curatorFramework.create()
-                .withMode(CreateMode.EPHEMERAL_SEQUENTIAL)
+        curatorFramework.create()
+                .withMode(CreateMode.CONTAINER)
```
Comment (szczygiel-m) [inline hermes-common/src/main/java/pl/allegro/tech/hermes/common/admin/zookeeper/ZookeeperAdminTool.java:39]:
```suggestion
                .withMode(CreateMode.EPHEMERAL_SEQUENTIAL)
```
I have verified that all you have to do is change it, and the tests pass 😄

**Review by szczygiel-m — CHANGES_REQUESTED** [review]

File: hermes-common/src/main/java/pl/allegro/tech/hermes/common/admin/zookeeper/ZookeeperAdminTool.java (hunk 1 above)
Comment (kirsteend) [reply hermes-common/src/main/java/pl/allegro/tech/hermes/common/admin/zookeeper/ZookeeperAdminTool.java:39]: Thank you so much @szczygiel-m 👏 
Let me apply that change

**Review by szczygiel-m — APPROVED** [review]

**Review by pitagoras3 — APPROVED** [review]

**Pull request merged**